src/AppRoot.tsx

```diff
@@ -456,6 +472,55 @@ class AppRoot extends React.Component<Props, State> {
     return true
   }
 
+  public startStatusPolling = () => {
+    console.log('start status polling', {
+      statusPoller: this.statusPoller,
+      hardware: this.props.hardware,
+    })
+    /* istanbul ignore else */
+    if (!this.statusPoller) {
+      console.log('creating statusPoller…')
+      this.statusPoller = IntervalPoller.start(
+        GLOBALS.CARD_POLLING_INTERVAL,
+        async () => {
+          try {
+            // Possible implementation
+            const {
+              readAccesssibleControllerStatus,
+              readBatteryStatus,
+              readCardReaderStatus,
+              readPrinterStatus,
+            } = this.props.hardware
+            const accesssibleController = await readAccesssibleControllerStatus()
```

Comment:
This is the call that’s making `this` be `undefined` inside `readAccesssibleControllerStatus`. When you call a function without a context in strict mode you get `this === undefined` (in non-strict mode you get `this === window`).

For example:
```ts
const obj = { m() { return this } }
const { m } = obj
const mBound = ::obj.m // https://github.com/tc39/proposal-bind-operator

// These all pass.
assert(obj.m() === obj)
assert(m() === undefined)
assert(m.call(obj) === obj)
assert(mBound() === obj)
```

Note that the bind operator proposal is still stage 0 and [won’t be implemented in TS until stage 3 at the earliest](https://github.com/Microsoft/TypeScript/issues/3508).